include/boost/http_proto/rfc/detail/rules.hpp

```diff
@@ -229,6 +230,35 @@ status_line_rule =
 //------------------------------------------------
 
 // header-field   = field-name ":" OWS field-value OWS
+struct field_name_rule_t
+{
+    using value_type = core::string_view;
+
+    system::result<value_type>
+    parse(
+        char const *&it,
+        char const *end) const noexcept;
+};
+
+constexpr field_name_rule_t field_name_rule{};
+
+struct field_value_rule_t
+{
+    struct value_type
+    {
+        core::string_view value;
+        bool has_obs_fold = false;
+        bool has_crlf = false;
```

Comment:
these need at least a little programmer documentation. what does "has_crlf" mean? is this in addition to has_obs_fold or in lieu of it?